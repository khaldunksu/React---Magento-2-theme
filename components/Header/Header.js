import React, {Component} from "react";
import PropTypes from "prop-types";
import Navigation from "../Navigation/Navigation";
import {connect} from "react-redux";
import translate from "../../translate-maker";
import "./Header.scss";
import {stores} from "../../project.config";
import { actionTypes } from "../../store/store";
import { Router } from "../../routes";


export class Header extends Component {
    constructor(props) {
        super(props);

        this.changeLanguage = this.changeLanguage.bind(this);

    }

    async changeLanguage(ev) {
        let value = ev.target.value;
        await this.props.dispatch({
            type: actionTypes.SET_LOCALE,
            data: value
        });

        if(value !=="default"){
            Router.pushRoute('index', {lang: value})
        } else{
            Router.pushRoute('index-default')
        }
    }

    render() {
        let {locale} = this.props;
        locale = locale ? locale : "default";

        return (
            <div className="header js-hook__header">
                <Navigation/>
                <form action="/catalogsearch/result/">
                    <input name="q" type="text" placeholder={translate[locale].searchLabel} className="js-hook__header-search"/>
                </form>
                <h1 className="js-hook__header-logo-title">{translate[locale].storeTitle}</h1>
                <div>
                    <a href="/customer/account/login" className="js-hook__header-login">{translate[locale].loginBadge}</a>
                    <a href="/customer/account/create" className="js-hook__header-register">{translate[locale].registerBadge}</a>
                    <a href="/checkout/cart" className="js-hook__header-cart">Cart Icon</a>


                    <select name="languages" onChange={this.changeLanguage} defaultValue={locale} className="js-hook__header-languages">
                        {Object.keys(stores).length ? Object.keys(stores).map((lang, i) => {

                            let localeLang = stores[lang].urlPrefix !== '' ? stores[lang].urlPrefix : "en";
                            return <option key={"lang" + i} value={lang}>{localeLang}</option>

                        }): null}
                    </select>
                </div>
            </div>

        )
    }

}

Header.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
};
Header.defaultProps = {
    children: null,
};

export default connect(state => state)(Header);