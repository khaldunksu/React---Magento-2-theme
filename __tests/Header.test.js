import { shallow } from "enzyme";
import React from "react";
import Router from "next/router";
import toJson from "enzyme-to-json";
import { Header } from "../components/Header/Header";

const mockedRouter = { push: () => {} };
Router.router = mockedRouter;

describe("Component Header matches snapshot", () => {
    it("renders page layout correctly", () => {
        const wrapper = shallow(<Header />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component Header", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Header />);
    });

    it("renders the dumb component", () => {
        expect(wrapper.length).toEqual(1);
        expect(wrapper.find(".js-hook__header").length).toBe(1);
    });

});