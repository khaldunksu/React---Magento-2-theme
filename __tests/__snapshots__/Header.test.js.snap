// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header matches snapshot renders page layout correctly 1`] = `
<div
  className="header js-hook__header"
>
  <Navigation />
  <form
    action="/catalogsearch/result/"
  >
    <input
      className="js-hook__header-search"
      name="q"
      placeholder="Search"
      type="text"
    />
  </form>
  <h1
    className="js-hook__header-logo-title"
  >
    Superheroes
  </h1>
  <div>
    <a
      className="js-hook__header-login"
      href="/customer/account/login"
    >
      Log in
    </a>
    <a
      className="js-hook__header-register"
      href="/customer/account/create"
    >
      Registration
    </a>
    <a
      className="js-hook__header-cart"
      href="/checkout/cart"
    >
      Cart Icon
    </a>
    <select
      className="js-hook__header-languages"
      defaultValue="default"
      name="languages"
      onChange={[Function]}
    >
      <option
        key="lang0"
        value="default"
      >
        en
      </option>
      <option
        key="lang1"
        value="ru"
      >
        ru
      </option>
      <option
        key="lang2"
        value="ua"
      >
        ua
      </option>
    </select>
  </div>
</div>
`;
